cfgov/v1/documents.py

```diff
@@ -152,11 +152,20 @@ def filter_language(self, language=None):
             )
 
     def filter_date(self, from_date=None, to_date=None):
-        if to_date is not None and from_date is not None:
-            self.search_obj = self.search_obj.filter(
-                "range",
-                **{"date_published": {"gte": from_date, "lte": to_date}},
-            )
+        if from_date and to_date:
+            ranges = [
+                {"end_date": {"gte": from_date}},
+                {"start_date": {"lte": to_date}},
+            ]
+        elif from_date:
+            ranges = [{"end_date": {"gte": from_date}}]
+        elif to_date:
+            ranges = [{"start_date": {"lte": to_date}}]
+        else:
+            ranges = []
```

Comment:
@csebianlander I tried to write tests for this but I would appreciate your once-over on this logic to make sure I did this right. What I tried to do here was make the ranges inclusive; in other words, say I have an event that starts at 12PM and ends at 1PM. If I search with a range from START to END, then END has to be after 12PM, and start has to be before 1PM. If I give only START, then START has to be before 1PM (my range overlaps with the end of the event) or, if I give only END, then END has to be after 12PM (my range overlaps with the beginning of the event). Does this seem right to you?

Reply:
yeah that seems right to me!